pkg/web/api.go

```diff
@@ -82,15 +84,158 @@ func (api *API) RegisterFilters(filters ...Filter) {
 	api.Filters = append(api.Filters, filters...)
 }
 
+func (api *API) registerFilterRelatively(filterName string, newFilter Filter, newFilterPosition func(filterPosition int) int) {
+	registeredFilterPosition := api.findFilterPositionOrDie(filterName)
+	filterPosition := newFilterPosition(registeredFilterPosition)
+	api.Filters = append(api.Filters, nil)
+	copy(api.Filters[filterPosition+1:], api.Filters[filterPosition:])
+	api.Filters[filterPosition] = newFilter
+}
+
+func (api *API) RegisterFilterBefore(beforeFilterName string, filter Filter) {
+	logrus.Debugf("Registering filter %s before %s", filter.Name(), beforeFilterName)
+	api.registerFilterRelatively(beforeFilterName, filter, func(beforeFilterPosition int) int {
+		if beforeFilterPosition == 0 {
+			return 0
+		}
+		return beforeFilterPosition - 1
+	})
+}
+
+// RegisterFilterAfter adds the given filter after the filter with the specified name
```

Comment:
RouteMatchersValue []FilterMatcher of pluginsegment - make it single filter matcher